src/robot/libraries/BuiltIn.py

```diff
@@ -613,10 +613,21 @@ def should_be_equal(self, first, second, msg=None, values=True):
         for example, string ``false`` or ``no values``. See `Boolean arguments`
         section for more details.
 
+        If ignore_case is True, it indicates that ``first`` and ``second`` should be
+        compared case-insensitively.  See `Boolean  arguments` section for more details.
+        (This option is new in Robot Framework 3.0.1)
+
         If both arguments are multiline strings, the comparison is done using
         `multiline string comparisons`.
         """
         self._log_types_at_info_if_different(first, second)
+        if is_truthy(ignore_case):
+            if is_string(first) and is_string(second):
+                first = first.lower()
+                second = second.lower()
+            elif is_list_like(first) and is_list_like(second):
+                first = [x.lower() if type(x) is str else x for x in first]
+                second = [x.lower() if type(x) is str else x for x in second]
```

Comment:
I don't think it is a good idea to compare list-like items case-insensitively with this keyword in general. Even if it would be, the current implementation considers a list `['foo', 'bar']` equal to a  tuple `('foo', 'bar')` and even to a dict `{'foo': 1, 'bar': 2}`. Additionally, the current implementation wouldn't handle nested lists like `[['foo']]` and `[['FOO']]`.

Better to simply remove the `elif` block and just use `if is_truthy(ignore_case) and is_string(first) and is_string(second)`. If we want to make it explicit that `ignore_case=True` only applies to strings, the docstring could be changed to something like this:

> If ``ignore_case`` is given a true value (see `Boolean arguments` for details) and ``first`` and ``second`` are strings, the comparison is done case-insensitively.